### DISCUSSION TITLE
[Feature Request] Reduced Server Communication - Commits

### DISCUSSION THREAD

**Discussion by ZebTheWizard** [livewire/livewire#1528] [4. General]
### Scenario:
I have a settings page full of checkboxes and inputs that I want to sync to the database. Something as simple as rapidly checking and unchecking a box would significantly hog network bandwidth. 

### Problem
Big websites cannot use Livewire because there are too many network requests. 

### My Proposals:

1. **Throttling**
If a user reaches a throttle limit then Livewire would throw an error and intelligently waits before trying again. No further network requests would occur until the throttle limit has expired. 

2. **Commits (Transactions)**
There should be a way to keep all state changes in the browser. Then, when a method like "commit" is called, all previous state changes are sent to the server. I propose a setting in the config file or some sort of trait that allows each component to have commit functionality.

### Real Example
Discord's settings page is a prime example of what commit functionality looks like. Users can make all the changes they want, but settings are only sent to the server upon final confirmation.
![image](https://user-images.githubusercontent.com/13527122/92776836-3c788800-f365-11ea-9773-0ebbe3c42ba4.png)

**Comment by Wulfheart** [comment]
What about using ``.defer``?

**Reply by redbastie** [reply]
^ what they said. use wire:model.defer="myAction"
